### PR TITLE
add user menu button to home screen

### PR DESCRIPTION
**Pull request by redreceipt**
Kapture 2021-06-28 at 23 04 55

### CODE REVIEW

**Comment by redreceipt** [comment]
this will need to wait on deciding what type of navigation it will be, there's a lot of subscreens inside the current user settings area

**Comment by redreceipt** [comment]
Figured this was an easy first step that will open up making the connect tab feature-driven. we can tweak the UX of the user settings if it's not quite right later. But I already love having the user menu on the home screen.

**Comment by conrad-vanl** [comment]
@redreceipt I think we should put the actual menu inside of a stack navigator, so we don't have endless modals opening?

**Comment by redreceipt** [comment]
we _could_, I started on that, but this is simpler code. and it's only max two modals. and this doesn't feel bad to me. but definitely up to you

**Comment by conrad-vanl** [comment]
Did you test on android? No swipe gestures on android, I know there's a hardware back button, but since we don't render one ourselves in any of these views it could be funky.

What about using a drawer? https://reactnavigation.org/docs/drawer-based-navigation

**Comment by redreceipt** [comment]
O shoot, good call. Had no idea there’s no swipe on Android. I’ll add buttons. Is it not weird that search opens a modal and the user would open a drawer? That’s another reason I went with modal.

**Comment by redreceipt** [comment]
ok @conrad-vanl I added close modal buttons to the local components, we can add them to the core components in an `apps` PR, it's a current bug anyway (can't get out of `Location`) without selecting confirm.

**Comment by redreceipt** [comment]
@conrad-vanl anything keeping you from approving this? You dead set on drawer for user menu?

**Comment by conrad-vanl** [comment]
It's on my list to run and test I haven't done that yet. Also kicking off @rkdavidson tomorrow and there's a couple of PRs I might suggest he starts getting involved in. 

I'm not dead set on it being a drawer. I would really like there to either be a stack Nav or a drawer though 😏

**Comment by redreceipt** [comment]
Is it not weird to you have a stack on one side and a modal on the other (search)?

**Comment by conrad-vanl** [comment]
I don't dislike the modal. I dislike the stacked modals. 

Search button opening a modal wasn't original solution either

**Comment by redreceipt** [comment]
So what you make search ideally? Also stack?

**Comment by conrad-vanl** [comment]
Not sure. 

I like the modal like you have it a lot, and it'd be killer in a stack nav 😉

**Comment by conrad-vanl** [comment]
Both a modal like you have it is great. Just the sub links on the profile instead of opening a second modal would ideally be a stack nav similar to opening sibling content items

**Comment by redreceipt** [comment]
ok @conrad-vanl theres your stack wrapped in a modal. have a look.

**Comment by redreceipt** [comment]
no idea why these snapshots aren't matching mine locally

**Comment by redreceipt** [comment]
> no idea why these snapshots aren't matching mine locally

yarn cache...

**Comment by rkdavidson** [comment]
Still getting my local env up and running so I can familiarize with the UX here, but FWIW I agree with @conrad-vanl about preferring a stack nav _within_ the menu modal, for sub-settings screens.

And feels like we need a header on the sub-screens to anchor the content. That'd make it clear what sub-section you're in, and make room for an explicit back button.

**Comment by redreceipt** [comment]
Awesome. When you pull down my latest updates, it’s much closer to what you’re describing. Only thing I’d need to add is a title somehow.

**Comment by redreceipt** [comment]
@conrad-vanl @rkdavidson wanna give this one a look? vineyard is gonna need it

**Comment by rkdavidson** [comment]
Ah sorry @redreceipt , I still can't get my local apollos app stable to test this 😩 .
I'll set aside time tomorrow or next week to pair up with somebody on the team who can help, but spinning wheels playing whack-a-mole with random environment dependency and config issues.

**Comment by redreceipt** [comment]
Can I get a look at this? I can help debug environment issues if necessary.

**Comment by conrad-vanl** [comment]
So what happens to the connect tab in this version? Is it a precursor to making connect tab feature driven?

**Comment by redreceipt** [comment]
yes that will be in another PR. this makes the connect tab action bar, action table, and follow sections

**Comment by rkdavidson** [comment]
@redreceipt & @conrad-vanl I don't think I've got much value to add on this PR right now, but, happy to take @redreceipt up on his offer to get my env issues resolved if you guys disagree!

**Comment by nlewis84** [comment]
When I made this on my own, I did a stack inside a modal, so I'm happy to see that here! Having the search and avatar both open  a modal makes sense to me.

**Comment by nlewis84** [comment]
There's still some UX stuff to iron out here and the cog on the Connect tab needs to go away. This video is where it stands currently.

https://user-images.githubusercontent.com/72768221/131373747-45ea8a8b-7c45-4fe9-9de1-60ed3a078d36.mp4

**Comment by redreceipt** [comment]
thanks @nlewis84 . here's a PR to fix the duplicate avatars: https://github.com/ApollosProject/apollos-apps/pull/2154

I'll get the UI cleaned up

**Comment by conrad-vanl** [comment]
Here's a mockup of what I'd like this screen to look like:

Screen Shot 2021-08-30 at 1 47 04 PM

[Figma Artboard](https://www.figma.com/file/8zgpqqcJ79VIZF6dRBlBe6/Multi-Platform-Exploration?node-id=252%3A7414)

As for child screens - notice that I include the native navigation header on this mockup instead of our own custom Modal Header. I think we can replicate the UX you see when you tap on the profile avatar in the App Store app, and have these child pages just use the built-in stack navigation and header. That might fix some of the other UX issues seen on the other screens in the video above.

Docs for the native stack navigator (and header props) are here - https://github.com/software-mansion/react-native-screens/tree/master/createNativeStackNavigator

**Comment by conrad-vanl** [comment]
Actually, would would be sweet is if we could replicate what you see on the Game Center screen:

https://user-images.githubusercontent.com/103927/131389804-0b0bc8ec-717e-4cda-a578-2b42fdde5ca5.MP4

I think we could achieve the "floating header" affect from the Game Center screen by combining [`headerTranslucent`](headerTranslucent) and [`blurEffect`](https://github.com/software-mansion/react-native-screens/tree/master/createNativeStackNavigator#headerstyle). That would only be used on this first screen, child screens would revert to a standard opaque header.

**Comment by conrad-vanl** [comment]
I mocked up what I think this would look like in this prototype: https://www.figma.com/proto/8zgpqqcJ79VIZF6dRBlBe6/Multi-Platform-Exploration?page-id=0%3A1&node-id=252%3A7414&viewport=379%2C48%2C1.29&scaling=scale-down&starting-point-node-id=2%3A246&show-proto-sidebar=1

File: apolloschurchapp/ios/Podfile.lock [hunk 1]
```diff
@@ -613,7 +613,7 @@ SPEC CHECKSUMS:
   boost-for-react-native: 39c7adb57c4e60d6c5479dd8623128eb5b3f0f2c
   BVLinearGradient: e3aad03778a456d77928f594a649e96995f1c872
   CocoaAsyncSocket: 065fd1e645c7abab64f7a6a2007a48038fdc6a99
-  DoubleConversion: cde416483dac037923206447da6e1454df403714
+  DoubleConversion: cf9b38bf0b2d048436d9a82ad2abe1404f11e7de
```
Comment (redreceipt) [inline apolloschurchapp/ios/Podfile.lock:616]:
```suggestion
  DoubleConversion: cde416483dac037923206447da6e1454df403714
```

File: apolloschurchapp/ios/Podfile.lock [hunk 2]
```diff
@@ -624,7 +624,7 @@ SPEC CHECKSUMS:
   Flipper-RSocket: 127954abe8b162fcaf68d2134d34dc2bd7076154
   FlipperKit: 8a20b5c5fcf9436cac58551dc049867247f64b00
   Folly: b73c3869541e86821df3c387eb0af5f65addfab4
-  glog: 40a13f7840415b9a77023fbcae0f1e6f43192af3
+  glog: 73c2498ac6884b13ede40eda8228cb1eee9d9d62
```
Comment (redreceipt) [inline apolloschurchapp/ios/Podfile.lock:627]:
```suggestion
  glog: 40a13f7840415b9a77023fbcae0f1e6f43192af3
```

**Comment by redreceipt** [comment]
@nlewis84 you gonna finish this one out?

**Comment by nlewis84** [comment]
Yeah, I can work on this.

**Comment by conrad-vanl** [comment]
Also, @redreceipt the VCC app expects to use features on this screen:

Screen Shot 2021-08-31 at 8 44 11 AM

How hard will it be to add a feature feed to this screen in the future so we can more easily control what shows up in addition to these static links?

**Comment by redreceipt** [comment]
@conrad-vanl I thought follow stuff went on the connect tab, it will go on the user settings modal?

**Comment by nlewis84** [comment]
VCC does not have a Connect Tab.

**Comment by conrad-vanl** [comment]
I think we need to make the following stuff feature driven and the 5 tabs plus the user menu can each have a feature feed on it

**Comment by redreceipt** [comment]
Ah, so then to answer your question @conrad-vanl, adding a feature feed to the user settings modal will be a new core feature set. however if no connect tab is the rare exception, then I think we just throw follow stuff on user settings here as a custom user settings modal.

**Comment by conrad-vanl** [comment]
Yeah, I'm fine with that in short term although I'd prefer to get away from "custom" 😉

**Comment by nlewis84** [comment]
@conrad-vanl @redreceipt This is where this PR currently stands. There is a bug in the likes page that I just found.

https://user-images.githubusercontent.com/72768221/131545113-dd951f89-35f5-41ed-8c77-99b529f9c4f6.mp4

**Comment by conrad-vanl** [comment]
- [ ] Lets change "User Settings" heading to just say "Profile"
- [ ] Lets add the user full name below the avatar image like in the mockup above
- [ ] Can we make the icons use `colors.text.secondary` or `colors.neutral.gray2` instead of the same as the text color?

**Comment by nlewis84** [comment]
Simulator Screen Shot - iPhone 11 - 2021-08-31 at 13 47 09

**Comment by rkdavidson** [comment]
LGTM overall, good tweaks there in that last iteration 👍 .
I like this style popover/modal/thing.

Nitpick suggestion and up to you @conrad-vanl , but these section headings feel slightly crammed in.
WDYT about a _bit_ more padding above the labels, to push the sections a little farther apart?

**Comment by conrad-vanl** [comment]
Yeah @rkdavidson I'd like to modify the default styles of actiontable as a whole to better match what's in Figma, I agree. outside scope of this Pr though.

@nlewis84 my only thing before merging I think would be - instead of using the "teal" colors.secondary color that you are using now for the icons, I'd like to use the `colors.neutral.gray2` (https://github.com/ApollosProject/apollos-apps/blob/master/packages/apollos-ui-kit/src/theme/types/light.js#L14) which is a gray color.

**Comment by nlewis84** [comment]
Simulator Screen Shot - iPhone 11 - 2021-08-31 at 15 41 56

**Comment by nlewis84** [comment]
https://github.com/ApollosProject/apollos-apps/pull/2163 will fix the likedContentFeed bug.

**Comment by conrad-vanl** [comment]
FYI, here's an example of what I think the Connect tab can look like with the result of this PR:
image

**Comment by nlewis84** [comment]
Actually, there is a lot there. I can make that a separate PR. Also, to remove the LikedContentFeed, I'll need to an `apps` PR.

**Comment by conrad-vanl** [comment]
Sounds good @nlewis84 regarding separate PR.

For `containedModal` (GREAT find by the way!), did you see [this](https://github.com/software-mansion/react-native-screens/blob/master/native-stack/README.md#stackpresentation): 

> Using containedModal and containedTransparentModal with other types of modals in one native stack navigator is not recommended and can result in a freeze or a crash of the application.

**Comment by nlewis84** [comment]
https://user-images.githubusercontent.com/72768221/131744988-36cfa8a2-7715-41db-b64f-fa557c39b4fa.mp4

**Review by conrad-vanl — APPROVED** [review]
This is looking great! Need to make sure we test in Android before merging.... but other then that I think this is g2g

**Comment by nlewis84** [comment]
@Vincent Have an idea what would make the Avatar unclickable on Android only? Navigating to the UserSettingsNavigator from clicking the search Icon works, but no navigates from clicking the Avatar are working....actually not even console.logs are registering from clicking the avatar.

**Pull request merged**